app/controllers/api/categories_controller.rb

```diff
@@ -0,0 +1,18 @@
+class API::CategoriesController < API::BaseController
+  include Roar::Rails::ControllerAdditions
+  represents :json, entity: CategoryRepresenter
+
+  def index
+    respond_with main, represent_items_with: CategoryRepresenter
+  end
+
+  def show
+    respond_with Category.find(params[:id])
+  end
+
+  private
+
+  def main
```

Comment:
Just to shorten `Category.main.to_a`. It made more sense when `main` was `supercategories`, so I'll inline it again.